### PR TITLE
Use new condor env format

### PR DESCRIPTION
**Pull request by sthapa**
A small patch to use new condor env format

### CODE REVIEW

**Comment by brianhlin** [comment]
@edquist please make time to review this

File: src/scripts/condor_submit.sh [hunk 1]
```diff
@@ -201,19 +201,27 @@ submit_file_environment="#"
...
-            submit_file_environment="environment = "
+            submit_file_environment="environment = \" ${env_var}"
         else
-            submit_file_environment="$submit_file_environment;"
+            # check for spaces in env_var
+            pattern=" "
+            if [[ $env_var =~ $pattern ]]; then
+              fixed_env_var="${env_var}'"
+              fixed_env_var=`echo ${fixed_env_var} | sed -e "s|=|='|"`
+              echo "12 -- ${fixed_env_var}" >> ~/foo
```
Comment (edquist) [inline src/scripts/condor_submit.sh:217]: This looks like a debugging line that should be removed.

**Comment by edquist** [comment]
What is the expected input format for the `-V` option? (that ends up in `$environment`)

Can it contain `'`'s or `"`'s?

BTW I notice in the usage string it mentions `[-v <environment>]`, instead of `-V`, which is the one this change deals with.  Not sure if this detail is relevant.

File: src/scripts/condor_submit.sh [hunk 2]
```diff
@@ -201,19 +201,27 @@ submit_file_environment="#"
...
-# FIXME: probably it's better to convert everything into the 'new' Condor
-# environment format.
+# new condor format to avoid errors  when things like LS_COLORS (which 
+# has semicolons in it) get captured
     eval "env_array=($environment)"
     submit_file_environment=""
     for  env_var in "${env_array[@]}"; do
         if [ "x$submit_file_environment" == "x" ] ; then
-            submit_file_environment="environment = "
+            submit_file_environment="environment = \" ${env_var}"
```
Comment (edquist) [inline src/scripts/condor_submit.sh:210]: Probably minor, but is the space here between `\"` and `${env_var}"` intentional?  I'd guess no but it seems worth asking.

File: src/scripts/condor_submit.sh [hunk 3]
```diff
@@ -201,19 +201,27 @@ submit_file_environment="#"
...
-            submit_file_environment="$submit_file_environment;"
+            # check for spaces in env_var
+            pattern=" "
+            if [[ $env_var =~ $pattern ]]; then
+              fixed_env_var="${env_var}'"
+              fixed_env_var=`echo ${fixed_env_var} | sed -e "s|=|='|"`
+              echo "12 -- ${fixed_env_var}" >> ~/foo
+              submit_file_environment="${submit_file_environment} ${fixed_env_var} "
+            else
+              submit_file_environment="${submit_file_environment} ${env_var} "
```
Comment (edquist) [inline src/scripts/condor_submit.sh:220]: Does this handle the semicolons issue mentioned in the comment about `LS_COLORS`?

Seems like the escaping happens only when there are spaces.

**Comment by edquist** [comment]
I've made a PR against @sthapa's branch with my suggested cleanups: https://github.com/sthapa/BLAH/pull/1

If i understand what's going on correctly, this should do the same thing, but maybe in a bit more straightforward way, and also handle semicolons, which as (i read the comments) was apparently the intention.

It does not try to handle embedded single-quotes (but neither does the original code).  If we want that, i suppose it'd just be an extra line to quote those too, but i don't know what the expected quoting rules are / how/where this variable gets used down the line.

**Review by edquist — CHANGES_REQUESTED** [review]
See comments.

Remove the debugging line at a minimum, and consider suggested cleanups.

File: src/scripts/condor_submit.sh (hunk 1 above)
Comment (sthapa) [reply src/scripts/condor_submit.sh:217]: Yeah, that is debugging output, can be removed

File: src/scripts/condor_submit.sh (hunk 2 above)
Comment (sthapa) [reply src/scripts/condor_submit.sh:210]: I don't believe it is, but I don't think it hurts.

File: src/scripts/condor_submit.sh (hunk 3 above)
Comment (sthapa) [reply src/scripts/condor_submit.sh:220]: Yes, it does.  The new condor env format doesn't use semicolons as delimiters.  The problem was the original env format used semicolons e.g. env=FOO=bar;TERM=xterm;OSG_SQUID=blah   .  With the new format spaces are delimiters so we need to escape them but that should be it.

**Comment by brianhlin** [comment]
@edquist we don't have to address this in the 3.4.24 release of the blahp but let's get this done for the following version. Submit a new PR to opensciencegrid/blahp (either based off of this one or a completely new one)

**Comment by edquist** [comment]
@sthapa, thanks for contributing this way back and I'm sorry for not getting it in sooner.  I did not really understand what the new condor env format was about at the time but after looking at the condor manual it makes sense to me now.  We'll get this PR merged in via #79 probably tomorrow with @brianhlin's approval.

**Comment by sthapa** [comment]
No worries, I'm sure there's enough things happening that this might have been put on the backburner for a while.  Happy to contribute.

**Pull request closed without merging**